Update Github Actions for the develop branch

optimisticben opened the pull request:
**Description**
Now that `develop` is the default branch, we should generate packages from the branch for testing deployments.

**Additional context**
Added the `develop` branch to the workflows:
- geth unit tests
- integration
- Release
- typescript / contracts

smartcontracts commented:
@optimisticben I think you might've built this PR on top of an older version of `develop`. We updated some of these files last night so there are a few conflicts that need to be resolved.

optimisticben commented:
> @optimisticben I think you might've built this PR on top of an older version of `develop`. We updated some of these files last night so there are a few conflicts that need to be resolved.

Thanks! Updated.
